docs/content/users/extend/traefik-router.md

```diff
@@ -21,10 +21,32 @@ All Traefik configuration uses the *file* provider, not the *Docker* provider. E
 
 Static configuration is automatically generated in the `~/.ddev/traefik` directory. "Static" configuration means Traefik configuration which is only read when the router is started.
 
-* `.static_config.yaml` is the base configuration, generated by DDEV, and not to be edited. It is read on router startup, and does not change after that.
-* Additional static configuration may be added by adding a `static_config.*.yaml` file, which will be appended to the generated `.static_config.yaml`.
+* `.static_config.yaml` (a hidden file) is the configuration that gets used. It is not to be edited; it is generated from DDEV's base configuration while merging any files named `static_config.*.yaml`. It is read on router startup, and does not change until the router starts again (normally after `ddev poweroff`).
+* Additional static configuration may be added by adding `static_config.*.yaml` files, which will be merged into the generated `.static_config.yaml`. For example, a `static_config.plugin.yaml` might contain external Traefik plugins, or a `static_config.dnschallenge.yaml` might provide configuration for additional `certificatesResolvers`. Merging is done with an **override** strategy, meaning that the last file (highest alphanumeric filename) to touch a particular element of the YAML structure wins.
+    Some examples of `static_config.*.yaml` files are:
+    * `static_config.cloudflare.yaml`:
+
+      ```yaml
+      certificatesResolvers:
+        acme-dnsChallenge:
+          acme:
+            email: admin@example.com
+            dnsChallenge:
+              provider: cloudflare
+      ```
+
+    * `static_config.fail2ban.yaml`
+
+        ```yaml
+        experimental:
+          plugins:
+            fail2ban:
+              moduleName: "github.com/tomMoulard/fail2ban"
+              version: "v0.8.1"
+      ```
+
 * `certs/default_cert.*` files are the default DDEV-generated certificates, normally created by `mkcert`.
-* `config/default_config.yaml` contains global dynamic configuration, including pointers to the default certificates. It is possible to add other Traefik configuration in the `config` directory, which will apply to all projects. For example, a `config/router_middlewares.yaml` file might provide middleware implementations that would apply to all projects.
+* `config/default_config.yaml` contains global *dynamic* configuration, including pointers to the default certificates. It is possible to add other Traefik configuration in the `config` directory, which will apply to all projects. For example, a `config/router_middlewares.yaml` file might provide middleware implementations that would apply to all projects.
```

Comment:
> Do these get merged in the same fashion as the static_config.*.yaml files? If so, should they follow the same xyz.*.yaml format?

No, no merging, they're part of the dynamic configuration, and nothing done with them. I didn't really realize how/why they might be useful until talking to you one day.

> Finally, if they are defined here, can they be overridden by the project-specific dynamic config? I assume so, but it might be worth stating it.

I think it's probably alphabetic. You can easily see what goes on by `docker exec -it ddev-router bash` and `cd config`. You'll see these default_config files there. You can edit them. My bet is that `ls -l` in that directory is meaningful.  But I also bet they only do things nobody else does.

I'm not completely sure why we ended up with these default_config, this isn't specifically a Traefik idea, seems to be a DDEV idea. Normally only has the default certs in it, and normally the default certs aren't used.